src/main/java/org/sitenv/ccdaparsing/model/CCDAFamilyHxObs.java

```diff
@@ -93,4 +93,83 @@ public CCDADataElement getCausedDeathReferenceText() {
 	public void setCausedDeathReferenceText(CCDADataElement causedDeathReferenceText) {
 		this.causedDeathReferenceText = causedDeathReferenceText;
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + ((ageOnSetReferenceText == null) ? 0 : ageOnSetReferenceText.hashCode());
+		result = prime * result + ((ageOnSetValue == null) ? 0 : ageOnSetValue.hashCode());
+		result = prime * result + ((causedDeathReferenceText == null) ? 0 : causedDeathReferenceText.hashCode());
+		result = prime * result + ((causedDeathValue == null) ? 0 : causedDeathValue.hashCode());
+		result = prime * result + ((observationTime == null) ? 0 : observationTime.hashCode());
+		result = prime * result + ((observationTypeCode == null) ? 0 : observationTypeCode.hashCode());
+		result = prime * result + ((observationValue == null) ? 0 : observationValue.hashCode());
+		result = prime * result + ((referenceText == null) ? 0 : referenceText.hashCode());
+		result = prime * result + ((statusCode == null) ? 0 : statusCode.hashCode());
+		result = prime * result + ((templateIds == null) ? 0 : templateIds.hashCode());
```

Comment:
@kushalvermaecw  this will increase the cognitive complexity of the method. We can delegate this check to a util method and call it here.